mplc/multi_partner_learning.py

```diff
@@ -60,594 +53,427 @@ def __init__(self,
+            raise ValueError('Only one partner is provided. Please use the dedicated SinglePartnerLearning class')
+
+    def fit_epoch(self):
+        # Clear Keras' old models
+        clear_session()
+
+        # Split the train dataset in mini-batches
+        self.split_in_minibatches()
+
+        # Iterate over mini-batches and train
+        for i in range(self.minibatch_count):
+            logger.info(f"(seq-final-agg) Minibatch n°{self.minibatch_index} of epoch n°{self.epoch_index}, "
+                        f"init model with a copy of the global model")
+            self.minibatch_index = i
+            self.fit_minibatch()
+
+        # At the end of each epoch, aggregate the models
+        self.model_weights = self.aggregator.aggregate_model_weights()
+
+
+class SequentialAverageLearning(SequentialLearning):
+    def __init__(self,
+                 partners_list,
+                 epoch_count,
+                 minibatch_count,
+                 dataset,
+                 aggregation=UniformAggregator,
+                 is_early_stopping=True,
+                 is_save_data=False,
+                 save_folder="",
+                 init_model_from="random_initialization",
+                 use_saved_weights=False,
+                 ):
+        super(SequentialAverageLearning, self).__init__(partners_list,
+                                                        epoch_count,
+                                                        minibatch_count,
+                                                        dataset,
+                                                        aggregation,
+                                                        is_early_stopping,
+                                                        is_save_data,
+                                                        save_folder,
+                                                        init_model_from,
+                                                        use_saved_weights)
+        if self.partners_count == 1:
+            raise ValueError('Only one partner is provided. Please use the dedicated SinglePartnerLearning class')
+
+    def fit_epoch(self):
+        # Clear Keras' old models
+        clear_session()
+
+        # Split the train dataset in mini-batches
+        self.split_in_minibatches()
+
+        # Iterate over mini-batches and train
+        for i in range(self.minibatch_count):
+            logger.info(f"(seqavg) Minibatch n°{self.minibatch_index} of epoch n°{self.epoch_index}, "
+                        f"init model with a copy of the global model")
+            self.minibatch_index = i
+            self.fit_minibatch()
+
+            # At the end of each minibatch, aggregate the models
+            self.model_weights = self.aggregator.aggregate_model_weights()
 
 
 def init_multi_partner_learning_from_scenario(scenario, is_save_data=True):
-    mpl = MultiPartnerLearning(
+    mpl = scenario.multi_partner_learning_approach(
```

Comment:
It feels a bit weird to have this construction method inbetween different MPL approaches

Reply:
Yes I agree. 
Unfortunately as Python doesn't allow multi constructors, I personally don't see any other solution that this one. 
Any idea ?

Reply:
In fact a mpl object needs a list of partner. The partners are only usable after being instancied in a scenario (instanciation which split the data between the partner). So we will always need a scenario object somehow. But we can want the parameter of the mpl to be different from those set in the scenario object. I suggest to allow this kind of changes via the **kwargs

Reply:
What I meant was not to not have it, I remember why we introduced it and I think it makes perfect sense. I meant that it would better to have either before or after all MPL child classes, not in the middle of them

Reply:
aaah ok
By the time I removed this function ... what do you think ?